### PR TITLE
Qde2

### LINKED ISSUE #10: Update readme
## Description:
- Update readme.md based on new WebV functionality

### LINKED ISSUE #14: Additional Parameter Checks
## Description
- Web Validate should validate the parameters and return a parse fail error if the parameters are not valid
  - Known example
    - array count = -1 - defaults to zero - should return array
- Update readme with parameter validation rules
- Update readme - if status code validation fails, no other rules are checked (by design)

## Expected Behavior
- Fail on invalid parameters
  - Display first 10 validation errors
- Content comparisons are case sensitive

## Additional Context
- Update model per design review
  - [X] Url --> Path
  - [X] Code --> StatusCode
  - [X] Add Length
  - [X] Remove JsonArray.CountIsZero

- Validate the following parameters per readme
  - [X] Path
  - [X] Verb
  - [X] StatusCode
  - [x] ContentType
  - [x] Length
  - [x] MinLength
  - [x] MaxLength
  - [x] MaxMilliSeconds
  - [x] ExactMatch
  - [ ] PerfTarget

## Doc Changes
- Updates to readme - References #10

### PR DESCRIPTION
**Pull request by mag16**
## Type of PR

- [X] Documentation changes
- [X] Code changes
- [X] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
- Update model per design review
  - [X] Url --> Path
  - [X] Code --> StatusCode
  - [X] Add Length
  - [X] Remove JsonArray.CountIsZero

- Validate the following parameters per readme
  - [X] Path
  - [X] Verb
  - [X] StatusCode
  - [x] ContentType
  - [x] Length
  - [x] MinLength
  - [x] MaxLength
  - [x] MaxMilliSeconds
  - [x] ExactMatch
  - [x] PerfTarget

## Doc Changes
- Updates to readme - References #10 

## Issues Closed or Referenced
- Closes #10 
- Closes #14

### CODE REVIEW

File: src/WebValidation/Model/Request.cs [hunk 1]
```diff
@@ -40,21 +41,42 @@ public bool Validate(out string message)
...
 
             // validate Length
+            if (Validation.Length != null || Validation.Length < 0)
+            {
+                return false;
+            }
 
             // validate MinLength
+            if (Validation.MinLength !=null || Validation.MinLength < 0)
+            {
```
Comment (aflinchb) [inline src/WebValidation/Model/Request.cs]: Minor thing -- I see that a few of these dont have a space between "!=" and null -- not sure if that will trigger a linting error in build.

File: src/WebValidation/Model/Request.cs [hunk 2]
```diff
@@ -40,21 +41,42 @@ public bool Validate(out string message)
...
+                return false;
+            }
 
             // validate MaxLength
+            if (Validation.MaxLength !=null || Validation.MaxLength < 0)
+            {
```
Comment (aflinchb) [inline src/WebValidation/Model/Request.cs]: Should this be Validation.MaxLength < MinLength (if it exists)? Or does that get checked elsewhere?

**Comment by aflinchb** [comment]
Meant to say! Be sure to add the issue that this PR is referencing to the PR description.  

References #18  (I think that's the issue)

File: src/WebValidation/Model/Request.cs (hunk 1 above)
Comment (bartr) [reply src/WebValidation/Model/Request.cs]: fixed - need to figure out why VS Code didn't catch this. It's not a linter error, so the code will compile but VS Code should catch the style error.

File: src/WebValidation/Model/Request.cs (hunk 2 above)
Comment (bartr) [reply src/WebValidation/Model/Request.cs]: updated

**Pull request merged**